torch/_dynamo/guards.py

```diff
@@ -639,7 +639,7 @@ def source_ref(source):
                 # TODO: we could make use of 'DefaultsSource' and offer a .guard.is_defaults() API
                 and "__defaults__" not in guard.name
                 and "__kwdefaults__" not in guard.name
-                and "hooks" not in guard.name
+                and (config.skip_nnmodule_hook_guards or "hooks" not in guard.name)
```

Comment:
@yf225 you can just delete this line (remove the 'hooks not in guard.name' logic) locally to unblock while we get this landed.